tools/castle-editor/code/framedesign.pas

```diff
@@ -2647,7 +2708,38 @@ procedure TDesignFrame.UpdateSelectedControl;
       for I := 0 to SelectedCount - 1 do
         SelectionForOI.Add(Selected[I]);
       for InspectorType in TInspectorType do
+      begin
         Inspector[InspectorType].Selection := SelectionForOI;
+        if not IsCollectionItemEditFormInitialized then
```

Comment:
Please move this section into a nested routine inside `TDesignFrame.UpdateSelectedControl;`, like `InitializePropertyEditorForms`. You can then move the new local variables (Row, J, Ed, Fm) there, and thus the `TDesignFrame.UpdateSelectedControl` implementation will be easier to read. Like

```pascal
procedure TDesignFrame.UpdateSelectedControl;

  procedure InitializePropertyEditorForms;
  var
    J: Integer;
    Row: TOIPropertyGridRow;
    Ed: TCollectionPropertyEditor;
    Fm: TCollectionPropertyEditorForm;
  begin
  end;

begin
  ...
end;
```

Reply:
Resolved in 0362e42